pkg/infra/basic/client.go

```diff
@@ -81,7 +81,10 @@ func DialConnection(node Node, logger *log.Logger) (*grpc.ClientConn, error) {
 	var connError error
 	var conn *grpc.ClientConn
 	for i := 1; i <= 3; i++ {
-		conn, connError = gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
+		conn, connError = gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) {
+			tlsConfig.InsecureSkipVerify = true
+			tlsConfig.ServerName = node.SslTargetNameOverride
```

Comment:
what if SslTargetNameOverride leave to empty?

Reply:
and I am not sure if we should add any kind of test as coverage?

Reply:
option.servername is a string so should be default to empty string

Good idea to have test coverage.
I will migrate my test back.